src/Http/Http/perf/Microbenchmarks/RequestTimeoutsMiddlewareBenchmark.cs

```diff
@@ -0,0 +1,149 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Http.Timeouts;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Http;
+
+public class RequestTimeoutsMiddlewareBenchmark
+{
+    RequestTimeoutsMiddleware _middlewareWithNoTimeout;
+    RequestTimeoutsMiddleware _middleware;
+    RequestTimeoutsMiddleware _middlewareWithThrow;
+
+    [GlobalSetup]
+    public void GlobalSetup()
+    {
+        _middlewareWithNoTimeout = new RequestTimeoutsMiddleware(
+            context => Task.CompletedTask,
```

Comment:
You'll get different results when the middleware goes async.
```suggestion
            context => Task.Yeild(),
```

Reply:
`Task.Yield()` returns `YieldAwaitable`, so I end up using `async context => { await Task.Yield(); },`

Here's the result:
|                           Method |     Mean |     Error |    StdDev |   Median |      Op/s |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|--------------------------------- |---------:|----------:|----------:|---------:|----------:|-------:|------:|------:|----------:|
|              NoMetadataNoDefault | 1.923 us | 0.0372 us | 0.0348 us | 1.919 us | 520,053.0 | 0.0458 |     - |     - |      1 KB |
|                   DefaultTimeout | 3.826 us | 0.0734 us | 0.0785 us | 3.802 us | 261,339.6 | 0.0076 |     - |     - |      2 KB |
| DefaultTimeoutOverridenByDisable | 3.029 us | 0.1516 us | 0.4470 us | 2.755 us | 330,172.5 | 0.0801 |     - |     - |      2 KB |
|                  TimeoutMetadata | 4.811 us | 0.0939 us | 0.0923 us | 4.831 us | 207,871.6 | 0.0153 |     - |     - |      3 KB |
|              NamedPolicyMetadata | 5.575 us | 0.2133 us | 0.6289 us | 5.239 us | 179,365.9 | 0.0076 |     - |     - |      3 KB |
|                     TimeoutFires | 4.953 us | 0.2131 us | 0.6283 us | 5.087 us | 201,890.6 | 0.0534 |     - |     - |      2 KB |